pkgs/development/libraries/harfbuzz/default.nix

```diff
@@ -43,24 +57,28 @@ stdenv.mkDerivation {
     (mesonFeatureFlag "graphite" withGraphite2)
     (mesonFeatureFlag "icu" withIcu)
     (mesonFeatureFlag "coretext" withCoreText)
+    (mesonFeatureFlag "introspection" withIntrospection)
   ];
 
+  strictDeps = true;
+
   nativeBuildInputs = [
     meson
     ninja
-    gobject-introspection
     libintl
     pkg-config
     python3
     gtk-doc
     docbook-xsl-nons
     docbook_xml_dtd_43
-  ];
+    glib # glib-mkenums
+  ] ++ lib.optional (withIntrospection) gobject-introspection;
 
   buildInputs = [ glib freetype cairo ] # recommended by upstream
+    ++ lib.optional (withIntrospection) gobject-introspection
     ++ lib.optionals withCoreText [ ApplicationServices CoreText ];
 
-  propagatedBuildInputs = []
+  propagatedBuildInputs = [ ]
     ++ optional withGraphite2 graphite2
```

Comment:
```suggestion
  propagatedBuildInputs = optional withGraphite2 graphite2
```

Reply:
I actually kinda like this style, easier to rearrange lines later.

Reply:
Done.